packages/grafana-prometheus/src/language_provider.ts

```diff
@@ -30,6 +31,13 @@ const EMPTY_SELECTOR = '{}';
 // Max number of items (metrics, labels, values) that we display as suggestions. Prevents from running out of memory.
 export const SUGGESTIONS_LIMIT = 10000;
 
+type UrlParamsType = {
+  start?: string;
+  end?: string;
+  'match[]'?: string;
+  limit?: string;
```

Comment:
Apologies if I'm mistaking what is going on here, but should this be gated on the version of Prometheus?
The `limit` parameter was only added in Prometheus 2.51; I don't think it complains if you add it before then, but users might be confused why it didn't work.

Reply:
You are corrrect @bboreham, thank you for the feedback! This should be gated for Prometheus 2.51.